src/lib/merkle_mask/masking_merkle_tree.ml

```diff
@@ -402,6 +402,17 @@ module Make (Inputs : Inputs_intf.S) = struct
       List.iter addresses_and_hashes ~f:(fun (addr, hash) ->
           self_set_hash t addr hash )
 
+    let set_merkle_path_unsafe t addr path =
+      assert_is_attached t ;
+      ignore
+        ( List.fold_left ~init:addr path ~f:(fun addr path ->
+              let sibling_addr = Location.Addr.sibling addr in
+              let addr = Location.Addr.parent_exn addr in
+              let hash = match path with `Left hash | `Right hash -> hash in
+              self_set_hash t sibling_addr hash ;
```

Comment:
I think we should use `addr` instead of `sibling_addr` here because merkle_path is already the path of siblings. We did it like this here because the merkle_path function itself is wrong.

Reply:
I cherry-picked all commits from PR #14586 (which I will close) to this branch and function `set_merkle_path_unsafe` stops being used and is completely removed by this change.

So I guess we can close this discussion.